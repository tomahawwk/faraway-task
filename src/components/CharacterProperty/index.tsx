import EditablePart, {IEditable} from 'components/EditablePart';
import {FC} from 'react';

export interface ICharacterProperty extends IEditable {
  delay: number;
  title: string;
}

const CharacterProperty: FC<ICharacterProperty> = ({
  onChange,
  property,
  delay,
  title,
}) => {
  return (
    <div
      className={`flex items-center text-[18px] gap-xs animation-fade-y animation-delay-${delay}`}>
      <div className="flex items-center gap-sm">
        <div className="w-[45px] h-[45px] relative rounded-full bg-grey">
          <img
            src={`images/${title}.svg`}
            className="absolute top-0 bottom-0 left-0 right-0 m-auto w-[22px]"
            alt={title}
          />
        </div>
        <span className="capitalize">{title}:</span>
      </div>
      <span className="text-primary-main">
        <EditablePart
          onChange={(e: React.ChangeEvent<HTMLDivElement>) => onChange(e)}
          property={property}
        />
      </span>
    </div>
  );
};

export default CharacterProperty;
