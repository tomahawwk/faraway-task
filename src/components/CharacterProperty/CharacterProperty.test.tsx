import {render, screen, fireEvent} from '@testing-library/react';
import CharacterProperty, {ICharacterProperty} from '.';

describe('CharacterProperty component', () => {
  const mockOnChange = jest.fn();
  const props: ICharacterProperty = {
    onChange: mockOnChange,
    property: 'property',
    delay: 2,
    title: 'Name',
  };

  test('renders property correctly', () => {
    render(<CharacterProperty {...props} />);
    const titleElement = screen.getByText('Name:');
    expect(titleElement).toBeInTheDocument();
  });

  test('calls onChange handler when EditablePart changes', async () => {
    render(<CharacterProperty {...props} />);
    const editablePartElement = screen.getByText('Name:');
    fireEvent.focus(editablePartElement);
    await fireEvent.keyUp(editablePartElement, {
      key: 'Enter',
      target: {innerText: 'New Value'},
    });
    setTimeout(() => {
      screen.findByText('New Value');
      expect(mockOnChange).toHaveBeenCalledWith({
        target: {innerText: 'New Value'},
      });
    }, 0);
  });

  test('CharacterProperty snapshot', () => {
    const characterProperty = render(<CharacterProperty {...props} />);

    expect(characterProperty).toMatchSnapshot();
  });
});
