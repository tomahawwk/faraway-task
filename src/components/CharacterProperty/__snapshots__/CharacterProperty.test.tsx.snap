// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterProperty component CharacterProperty snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center text-[18px] gap-xs animation-fade-y animation-delay-2"
      >
        <div
          class="flex items-center gap-sm"
        >
          <div
            class="w-[45px] h-[45px] relative rounded-full bg-grey"
          >
            <img
              alt="Name"
              class="absolute top-0 bottom-0 left-0 right-0 m-auto w-[22px]"
              src="images/Name.svg"
            />
          </div>
          <span
            class="capitalize"
          >
            Name
            :
          </span>
        </div>
        <span
          class="text-primary-main"
        >
          <div
            class="w-full flex gap-md items-center overflow-hidden"
          >
            <div
              class="outline-none ml-[-1px]"
              contenteditable="false"
            >
              property
            </div>
            <div
              class="flex duration-300 opacity-0"
            >
              <button
                class="bg-secondary-main text-white whitespace-nowrap pb-[6px] pt-[7px] duration-300
 px-xs rounded-xs uppercase font-medium tracking-wider text-[13px] hover:bg-grey"
              >
                Edit
              </button>
            </div>
          </div>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center text-[18px] gap-xs animation-fade-y animation-delay-2"
    >
      <div
        class="flex items-center gap-sm"
      >
        <div
          class="w-[45px] h-[45px] relative rounded-full bg-grey"
        >
          <img
            alt="Name"
            class="absolute top-0 bottom-0 left-0 right-0 m-auto w-[22px]"
            src="images/Name.svg"
          />
        </div>
        <span
          class="capitalize"
        >
          Name
          :
        </span>
      </div>
      <span
        class="text-primary-main"
      >
        <div
          class="w-full flex gap-md items-center overflow-hidden"
        >
          <div
            class="outline-none ml-[-1px]"
            contenteditable="false"
          >
            property
          </div>
          <div
            class="flex duration-300 opacity-0"
          >
            <button
              class="bg-secondary-main text-white whitespace-nowrap pb-[6px] pt-[7px] duration-300
 px-xs rounded-xs uppercase font-medium tracking-wider text-[13px] hover:bg-grey"
            >
              Edit
            </button>
          </div>
        </div>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
